listenbrainz/webserver/errors.py

```diff
@@ -232,3 +231,15 @@ def to_xml(self):
             with tag('error', code=self.api_error.code):
                 text(self.api_error.message)
         return '<?xml version="1.0" encoding="utf-8"?>\n' + yattag.indent(doc.getvalue())
+
+
+class ValidationError(Exception):
+    """ Error class for raising when the submitted payload does not pass validation.
+    Only use for code paths common to LB API, API compat & API compat deprecated.
+    Throw this error from an util method, capture it in each of the corresponding
+    views and re-raise the API dependent error.
+    """
```

Comment:
good documentation!
Maybe we could call this ListenValidationError (or IncomingListenValidationError? SubmittedListen?) instead, to make it a bit clearer to what we're validating.

Reply:
ListenValidationError  sounds good. will rename to that.

Reply:
Done.